### PR TITLE
Basic HTTP configuration parsing

### PR DESCRIPTION
**Pull request by theatrus**
Take a URL, and a sequence of parameters which are added to a POST
request.

This is a configuration segment only, as part of a multi-part pull request.

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -261,19 +266,42 @@ static int sink_callback(void* user, const char* section, const char* name, cons
...
-        syslog(LOG_WARNING, "Grevious state problem");
-        return 0;
-    }
 
-    sink_config_stream* config = (sink_config_stream*)sink_in_progress;
-    if (NAME_MATCH("binary")) {
-        return value_to_bool(value, &config->binary_stream);
-    } else if (NAME_MATCH("command")) {
-        config->stream_cmd = strdup(value);
+    if (sink_in_progress->type == SINK_TYPE_STREAM) {
```
Comment (smparkes) [inline src/config.c]: switch?

File: tests/test_config.c [hunk 2]
```diff
@@ -629,6 +629,12 @@ command=foo\n\
 [sink_stream_other]\n\
 binary=false\n\
 command=not cat\n\
+[sink_http_hi]\n\
+url=https://google.com\n\
```
Comment (smparkes) [inline tests/test_config.c]: Maybe example.com?

**Comment by smparkes** [comment]
lgtm

File: src/config.c (hunk 1 above)
Comment (theatrus) [inline src/config.c]: Agree, would be more readable

**Comment by sortalongo** [comment]
+1

**Pull request merged**